source/auth/auth.rst

```diff
@@ -683,6 +683,239 @@ mechanism
+        implicit region for ‘sts.amazonaws.com’.  Drivers will need to derive the region to use from the endpoint. The region 
+        is the second piece of a FQDN name. While all official AWS STS endpoints start with “sts.”, there are non-AWS hosted 
+        endpoints and test endpoints that will not follow this rule.
+
+================= =========
+Host              Region
+================= =========
+sts.amazonaws.com us-east-1
+first.second      second
+first             us-east-1
+================= ========= 
+
+`MongoCredential`_ Properties
+`````````````````````````````
+
+username
+	MAY be specified. The non-sensitive IAM access key.
+
+source
+	MUST be "$external". Defaults to ``$external``.
+
+password
+	MAY be specified. The sensitive IAM secret key.
+
+mechanism
+	MUST be "MONGODB-IAM"
+
+mechanism_properties
+	AWS_SESSION_TOKEN
+		Drivers MUST allow the user to specify an AWS session token for authentication with temporary credentials.
+
+IAM Credentials
+   If a username and password is provided drivers MUST use these for the IAM access key and IAM secret key, respectively. Drivers
+   MUST use this secret key to generate a signature using HMAC-SHA256.
+
+Temporary Credentials
+   If a username and password is not provided drivers MUST query the standard local AWS endpoint for temporary credentials.
+   If temporary credentials cannot be obtained then drivers MUST fail authentication and raise an error.
+
+   EC2 Instance
+      Calling the endpoint ``http://169.254.169.254/latest/meta-data/iam/security-credentials/`` from an EC2 instance 
+      will return the role attached to the EC2 instance in plaintext, if it exist:
+
+      ``role-name``
+
+      ``http://169.254.169.254/latest/meta-data/iam/security-credentials/<role-name>`` will return the following json response: 
+
+      .. code:: javascript
+
+            {
+                "Code": "Success",
+                "LastUpdated" : <date>,
+                "Type": "AWS-HMAC",
+                "AccessKeyId" : <access_key>,
+                "SecretAccessKey": <secret_access_key>,
+                "Token" : <security_token>,
+                "Expiration": <date>
+            }
+
+      .. note::
+
+      The IP address 169.254.169.254 is a link-local address and is valid only from the instance. 
+      See `Instance Metadata and User Data <https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html>`_.
+
+      From this response, drivers will need to extract the ``AccessKeyId``, ``SecretAccessKey`` and the ``Token`` argument values. 
+      Drivers will use these values to construct the call to ``sts::getCallerIdentity`` as part of the Amazon Signature V4 algorithm. 
+      Drivers who cache the security temporary credentials must check the expiration date before using them. 
+      If the credentials expire, they must refetch them. 
+
+   ECS Tasks
```

Comment:
How does a driver determine whether to use the `EC2 Instance` procedure or the `ECS Tasks` procedure for retrieving temporary credentials?

Reply:
I still don't understand how a driver is supposed to determine whether to use "EC2 Instance" or "EC2 Tasks". Do we try both?

Reply:
Added a section to address this. We'll just check if the environment variable AWS_CONTAINER_CREDENTIALS_RELATIVE_URI is populated or not.

Reply:
@ShaneHarvey please comment back or resolve.